### PR TITLE
[G2M] Architecture/defer long tasks

### PR DESCRIPTION
**Pull request by woozyking**

### CODE REVIEW

File: worker.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const diff = require('./modules/diff')
```
Comment (Goondrious) [inline worker.js]: it looked to me at first that the split of `.worker` and `.route` was for testing locally vs the actual lambda deploy, but I'm confused about this file

File: app.js [hunk 2]
```diff
@@ -3,21 +3,15 @@ const serverless = require('serverless-http')
...
 const axios = require('axios')
 
-const maintenance = require('./modules/maintenance')
-const deploy = require('./modules/deploy')
-const report = require('./modules/report')
+
+const { verifySlack } = require('./modules/middleware')
 const diff = require('./modules/diff')
 const food = require('./modules/food')
-const boardroom = require('./modules/boardroom')
```
Comment (Goondrious) [inline app.js]: why are these modules being removed?

File: modules/diff.js [hunk 3]
```diff
@@ -152,27 +149,48 @@ const getClientMeta = async (product) => {
...
+        console.error(err)
+        return res.status(200).json({ response_type: 'ephemeral', text: 'Failed to diff' })
+      }
+      return res.status(200).json({
+        response_type: 'ephemeral',
+        text: `Diffing for ${product}...`,
+      })
+    })
+  } else {
+    worker(payload).catch(console.error)
```
Comment (Goondrious) [inline modules/diff.js]: this mimics the immediate response + worker response, right?

**Comment by Goondrious** [comment]
It looks good to me, but had some clarifying questions. High level, this is what I see:

- trimmed some potentially unused modules
- each module now has `worker` and `route`
- each module immediately responds to the slack request, holding onto the `response_url` while processing is done and then sending actual data

File: modules/diff.js (hunk 3 above)
Comment (woozyking) [reply modules/diff.js]: yea, so `worker` is `async`, this works for local testing setup (or perpetually running server setup)

File: app.js (hunk 2 above)
Comment (woozyking) [reply app.js]: because they're not effective, or not used anymore

File: worker.js (hunk 1 above)
Comment (woozyking) [reply worker.js]: this is the collection of dedicated `worker` entrypoints specified through `serverless.yml`

Comment (Goondrious) [reply worker.js]: ooooh the reference is intact (in serverless it's `worker.xxx`)

**Comment by woozyking** [comment]
:ayyy:

Shane Stratton <notifications@github.com>于2020年1月13日 周一16:20写道：

**Pull request merged**